### ISSUE TITLE
Search for text within a Note - Jan 2021

### ISSUE THREAD

**Issue by deacl** [standardnotes/desktop#647]
We need to be able to search for text within a note and click to move focus from one match to the next, next, ....
I'm using desktop app on a Mac and now there is a search bar at the top which highlights all notes with matching the search text, but I still see no way to progress to the locations within the Note of each match.

I thought that this was promised earlier, is it planned?
thank you, -deacl

**Comment by arekm** [comment]
Agree. Searching is a nightmare in standard notes. Especially when each note is a few pages of text.

- it's hard to see which word is highlighted (I need to scroll *slowly* each note several times to notice highlighted word among tons of other words, numers etc)

- you never know how many times searched text was found in each note, could be one hit, could be tens etc. So you have to scroll carefully to do not miss any.

- when searching a text with space inside it finds some note that doesn't even have that text in it (but contains one word) for example searching for: nie można finds note (but doesn't highlight anything in it) which only has "nie" and "można" words in it but no "nie można" together. Maybe it treats: nie można as nie OR można but it doesn't highlight anything, so that's unknown.

**Comment by JaspalSuri** [comment]
Hi @deacl & @arekm, this feature is still planned as a part of v4, but we currently don't have an ETA on when v4 will be available. For the desktop & [web](https://app.standardnotes.org) apps, you can use the Monaco editor within the [Append Editor](https://appendeditor.com) to search for (and replace) content within your notes.
